src/irmin-pack/irmin_pack_layers.ml

```diff
@@ -727,49 +704,31 @@ struct
             X.Contents.CA.unsafe_consume_newies () |> List.rev |> Lwt.return
           else X.Contents.CA.consume_newies t.X.Repo.contents >|= List.rev
         in
-        let copy_contents contents t k =
-          X.Contents.CA.copy contents t.X.Repo.contents "Contents" k
-        in
-        let copy_tree contents nodes t k =
-          let skip_nodes k = mem_node_next t k in
-          let skip_contents k = mem_contents_next t k in
-          copy_tree ~skip_nodes ~skip_contents nodes contents t k
+        newies_commits >>= fun newies_commits ->
+        newies_nodes >>= fun newies_nodes ->
+        newies_contents >>= fun newies_contents ->
+        let newies_commits = List.rev_map (fun x -> `Commit x) newies_commits in
+        let newies_nodes = List.rev_map (fun x -> `Node x) newies_nodes in
+        let newies_contents = List.map (fun x -> `Contents x) newies_contents in
```

Comment:
this `List.map` also led to a stack overflow in my benchmarks, can you replace it with `rev_map`?